site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -6251,37 +6251,29 @@ private List getLtiToolGroup(String groupName, File moreInfoDir, Site site) {
 		if (allTools != null && !allTools.isEmpty()) {
 			for (Map<String, Object> tool : allTools) {
 				Set keySet = tool.keySet();
-				String toolIdString = tool.get(m_ltiService.LTI_ID).toString();
-				boolean toolStealthed = tool.get(m_ltiService.LTI_VISIBLE).toString().equals("1");
-				boolean ltiToolSelected = ltiSelectedTools.contains(toolIdString); 
-
-				try
-				{
-					// in Oracle, the lti tool id is returned as BigDecimal, which cannot be casted into Integer directly
-					Integer ltiId = Integer.valueOf(toolIdString);
-					if (ltiId != null) {
-						String ltiToolId = ltiId.toString(); 
-						if (ltiToolId != null && (!toolStealthed || ltiToolSelected) ) {
-							String relativeWebPath = null;
-							MyTool newTool = new MyTool();
-							newTool.title = tool.get("title").toString();
-							newTool.id = LTITOOL_ID_PREFIX + ltiToolId;
-							newTool.description = (String) tool.get("description");
-							newTool.group = groupName;
-							relativeWebPath = getMoreInfoUrl(moreInfoDir, ltiToolId);
-							if (relativeWebPath != null) {
-								newTool.moreInfo = relativeWebPath;
-							}
-							// SAK16600 should this be a property or specified in  toolOrder.xml?
-							newTool.required = false; 
-							newTool.selected = ltiToolSelected;
-							ltiTools.add(newTool);
-						}
-					}
-				}
-				catch (NumberFormatException e)
-				{
-					M_log.error(this + " Cannot cast tool id String " + toolIdString + " into integer value.");
+				Long ltiVisible = FoormUtil.getLong(tool.get(m_ltiService.LTI_VISIBLE));
+				Integer ltiId = FoormUtil.getInt(tool.get(m_ltiService.LTI_ID));
+
```

Comment:
I can look at this, I used FoormUtil as this is including the lti service already and FoormUtil is part of the basiclti api in shared so no new dependencies. These NumberUtils aren't exactly the same as they take a String parameterwhere FoormUtil.getLong can take an Object. In this case I don't know if you could guarantee that tool.get() returns a String as there aren't any defined classes here. I think all of this could be cleaned up a little more, but was just working on fixing the immediate NPE bug.